pyvista/core/utilities/geometric_sources.py

```diff
@@ -15,40 +15,65 @@
 from .helpers import wrap
 
 
-def translate(surf, center=(0.0, 0.0, 0.0), direction=(1.0, 0.0, 0.0)):
+def translate(
+    surf, center=(0.0, 0.0, 0.0), direction=(1.0, 0.0, 0.0), start_direction=(1.0, 0.0, 0.0)
+):
     """Translate and orient a mesh to a new center and direction.
 
-    By default, the input mesh is considered centered at the origin
-    and facing in the x direction.
+    The input mesh is considered centered at the origin
+    and facing in the start_direction.
 
     Parameters
     ----------
     surf : pyvista.core.pointset.PolyData
         Mesh to be translated and oriented.
-    center : tuple, optional, default: (0.0, 0.0, 0.0)
+    center : tuple, default: (0.0, 0.0, 0.0)
         Center point to which the mesh should be translated.
-    direction : tuple, optional, default: (1.0, 0.0, 0.0)
+    direction : tuple, default: (1.0, 0.0, 0.0)
         Direction vector along which the mesh should be oriented.
+    start_direction : tuple, default: (1.0, 0.0, 0.0)
+        Direction vector along which mesh is originally oriented.
 
     """
-    normx = np.array(direction) / np.linalg.norm(direction)
-    normy_temp = [0.0, 1.0, 0.0]
 
-    # Adjust normy if collinear with normx since cross-product will
-    # be zero otherwise
-    if np.allclose(normx, [0, 1, 0]):
-        normy_temp = [-1.0, 0.0, 0.0]
-    elif np.allclose(normx, [0, -1, 0]):
-        normy_temp = [1.0, 0.0, 0.0]
-
-    normz = np.cross(normx, normy_temp)
-    normz /= np.linalg.norm(normz)
-    normy = np.cross(normz, normx)
+    def _form_matrix(direction, inverse=False):
+        """Rotates from (1.0, 0.0, 0.0) to direction."""
+        normx = np.array(direction) / np.linalg.norm(direction)
+        normy_temp = [0.0, 1.0, 0.0]
+
+        # Adjust normy if collinear with normx since cross-product will
+        # be zero otherwise
+        if np.allclose(normx, [0, 1, 0]):
+            normy_temp = [-1.0, 0.0, 0.0]
+        elif np.allclose(normx, [0, -1, 0]):
+            normy_temp = [1.0, 0.0, 0.0]
+
+        normz = np.cross(normx, normy_temp)
+        normz /= np.linalg.norm(normz)
+        normy = np.cross(normz, normx)
+
+        out = np.vstack((normx, normy, normz))
+        if inverse:
+            return out
+        return out.transpose()
+
+    if np.array_equal(direction, start_direction):
+        rot_matrix = np.identity(3)
+    else:
+        # Start with inverse of matrix from start_direction to (1.0, 0.0, 0.0)
+        if not np.array_equal(start_direction, (1.0, 0.0, 0.0)):
+            rot_matrix = _form_matrix(start_direction, inverse=True)
+        else:
+            rot_matrix = np.identity(3)
+
+        # Multiply with matrix to direction to get final rotation matrix
+        rot_matrix = np.matmul(
+            _form_matrix(direction),
+            rot_matrix,
+        )
```

Comment:
While I'm here I'd expect this to be the same as
```suggestion
        rot_matrix = _form_matrix(direction) @ rot_matrix
```
unless you have a specific reason why not to use this.

Reply:
Yeah it is a great expression of NumPy :)